js/main.js

```diff
@@ -1,2 +1,232 @@
+        if (playerChoice === 'scissors') {
+          return 'bot'
+        } else if (playerChoice === 'paper') {
+          return 'player'
+        }
+        break
+
+      case 'paper':
+        if (playerChoice === 'rock') {
+          return 'bot'
+        } else if (playerChoice === 'scissors') {
+          return 'player'
+        }
+        break
+
+      case 'scissors':
+        if (playerChoice === 'paper') {
+          return 'bot'
+        } else if (playerChoice === 'rock') {
+          return 'player'
+        }
+        break
+      default:
+        console.log('Both players must choose an item!')
+        return 'nobody'
+    }
+  }
+
+  // update scoreboard page elements
+  function updateScore (winner) {
+    console.log(`running score count. adding points for ${winner}`)
+    // clear any bonus info from Scoreboard
+    $('#bonus').text('')
+    // get details from scoreboard
+    let currentHumanScore = parseInt($('#humanScore').text())
+    let currentBotScore = parseInt($('#computerScore').text())
+    // update the scoreboard and text with winner info
+    if (winner === 'player') {
+      currentHumanScore++
+      $('#showWinner').text('You win! :D')
+      $('#humanScore').text(currentHumanScore)
+    } else if (winner === 'bot') {
+      currentBotScore++
+      $('#showWinner').text('You lose :(')
+      $('#computerScore').text(currentBotScore)
+    }
+  }
+
+  // calculate bonus points for this turn
+  function calculateBonusPoints (winnerList) {
+    const n = winnerList.length - 1
+    // grab the last 3 winners from the winnerList
+    const one = winnerList[n]
+    const two = winnerList[n - 1]
+    const three = winnerList[n - 2]
+
+    // player who wins the last three rounds gets a two point bonus
+    if (one === two && two === three) {
+      if (one === 'nobody') {
+        console.log('no bonus for tie rounds.')
+      } else {
+        // add two bonus points and update scoreboard
+        updateScore(winnerList[n])
+        updateScore(winnerList[n])
+        $('#bonus').text(`Bonus: 2 pts for ${one}!`)
+      }
+    } else {
+      console.log('no bonus detected.')
+    }
+  }
+
+  // determine if there's a winner for the round
+  function calculateRoundWinner () {
+    //get score from page
+    const humanScore = parseInt($('#humanScore').text())
+    const botScore = parseInt($('#computerScore').text())
+
+    if (humanScore >= 10) {
+      // show a message that says Human wins the round!
+      // add a point for the round
+      alert('Congrats human, you\'ve won the round!')
+      updateRoundScore('player')
+    } else if (botScore >= 10) {
+      // show a message that says Bot wins the round!\
+      // add a point for the round
+      alert('Aww, you\'ve lost the round. :(')
+      updateRoundScore('bot')
+    } else {
+      // no one has won this round
+      // just in case
+      console.log('Round 1: in progress')
+    }
+  }
+
+  function updateRoundScore (winner) {
```

Comment:
@sad3xoxo the `pageUpdate` varibale is used quite often in this function and could be extended into a function that writes the new status with the respective winners passed as an argument. I read the comments, so great job anyway! 💯 

check out the approach [here](https://github.com/jsr20190826/rock-paper-scissors-solution/blob/1583be48146e2521b25308e017c52fbfcf0d9d89/js/main.js#L82)